cub/cub/device/device_partition.cuh

```diff
@@ -581,28 +641,18 @@ struct DevicePartition
      SelectSecondPartOp select_second_part_op,
      cudaStream_t stream = 0)
   {
-    using OffsetT                      = int;
-    using DispatchThreeWayPartitionIfT = DispatchThreeWayPartitionIf<
-      InputIteratorT,
-      FirstOutputIteratorT,
-      SecondOutputIteratorT,
-      UnselectedOutputIteratorT,
-      NumSelectedIteratorT,
-      SelectFirstPartOp,
-      SelectSecondPartOp,
-      OffsetT>;
-
-    return DispatchThreeWayPartitionIfT::Dispatch(
+    CUB_DETAIL_NVTX_RANGE_SCOPE_IF(d_temp_storage, "cub::DevicePartition::If");
+    return IfNoNVTX(
       d_temp_storage,
       temp_storage_bytes,
       d_in,
       d_first_part_out,
       d_second_part_out,
       d_unselected_out,
       d_num_selected_out,
+      num_items,
       select_first_part_op,
       select_second_part_op,
-      num_items,
       stream);
```

Comment:
Because `If` and `IfNoNVTX` have `num_items` as 4th last parameter, but `DispatchThreeWayPartitionIf::Dispatch` has `num_items` as the 2nd last parameter. Previously, the dispatch layer was called here, but I changed that to calling the new `IfNoNVTX`, so the argument had to move.